### PR TITLE
Added getMinOrMaxValueInArrayBySomeRules utility function

### PR DESCRIPTION
**Pull request by liuliangsir**
## Description

**Resolves** #(issue number) 

**Lodash[ACTION]** #100 -> https://lodash.com/docs/4.17.4#(method) 

## What does your PR belong to?
- [ ] Website
- [ ] Snippets
- [ ] General / Things regarding the repository (like CI Integration)
- [ ] Tests

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking improvement of a snippet)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## [Lodash Backlog](https://github.com/Chalarangelo/30-seconds-of-code/issues/100)

- [ ] I added the prefix [UPDATE: `method.md`] or [ADD: `method.md`]
- [ ] I have referenced the `method` to the lodash backlog.

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have checked that the changes are working properly
- [ ] I have checked that there isn't any PR doing the same
- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by skatcat31** [comment]
This is very close to calling tail on a sort function, but by using reduce you expose the single pass methodology people would want to use instead of sort. Great idea, but I think it should be called something shorter( will not add another code change request for the same @Chalarangelo has done )

HOW to name is the problem here though. I'd probably call it `reduceWhich` for the idea that you're reducing, but choosing Which to take along as a single value, but that's almost as confusing for new people as `liftA2` is to non FP people( still have to look that up myself even though I use it often -.- )

**Comment by skatcat31** [comment]
Another question is the footprint. I'm a fan of data last footprints, however since this takes a reducee and has a default function, keeping the footprint as is here is good enough. I can always use flip if I want to change it's footprint :wink:

**Comment by liuliangsir** [comment]
@skatcat31 

> will not add another code change request for the same @Chalarangelo has done

Give more details?  Because i am puzzled by the meaning of the upper statement and footprint

talking is cheap, show me your pull request

**Comment by skatcat31** [comment]
@liuliangsir 

# What I mean about the review
I'm not going to do a review( github term for if I approve the code at this moment or want changes ) with specifically asking for changes to the name as @Chalarangelo has already done the same review for the same reason and I agree that the only thing wrong with this snippet you submitted is the name. Other than that I think it's **great**! It's a good way to implement a single traversal reduction function to take a value based on the provided reducer. That's something most people would do a sort for instead of a reduce, something that is often overlooked for this exact scenario. While for me it's second nature to use a reduce here, it brings forth that it's actually better to use reduce so you have a small O(n) magnitude

# What I mean about the footprint
It was a comment on defaults for functions and personal taste, nothing more. It's more along the lines of me trying to be funny because I prefer
```js
const reduceWhich = (fn, args) => args.reduce( (a,b) => fn(a,b) >= 0 ? b : a )
```
Which allows me to do things like
```js
var whichOldest = reduceWhich.bind(null, (a,b) => b.age - a.age)
whichOldest([{name: 'Tom', age: 12}, {name: 'Jack', age: 18}, {name: 'Lucy', age: 9}]).name // 'Jack'
```
Which is worthless for me to ask for a change for( since I was asked for a review, I ussually tend to provide a full review ) since it's exactly the same as me doing
```js
whichOldest = flip(getMinOrMaxValueInArrayBySomeRules).bind(null, (a,b) => b.age - a.age)
whichOldest([{name: 'Tom', age: 12}, {name: 'Jack', age: 18}, {name: 'Lucy', age: 9}]).name // 'Jack'
```
using `flip`, one of our adapter functions.

Like I said, other than the name I think this is a great addition

**Comment by liuliangsir** [comment]
@skatcat31 @Chalarangelo rename name to `reduceWhich `, thx~

File: snippets/reduceWhich.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+reduceWhich([1, 3, 2]); // 1
+
+// get maximum value from numeric array
+reduceWhich([1, 3, 2], function (a, b) {return b - a;}); // 3
+
+// get minimum value from object array by a given key map to value
+reduceWhich([{name: 'Tom', age: 12}, {name: 'Jack', age: 18}, {name: 'Lucy', age: 9}], function(a, b){return a.age - b.age;}) // {name: "Lucy", age: 9}
+
+// get maximum value from object array by a given key map to value
+reduceWhich([{name: 'Tom', age: 12}, {name: 'Jack', age: 18}, {name: 'Lucy', age: 9}], function(a, b){return a.age - b.age;}) // {name: "Jack", age: 18}
```
Comment (skatcat31) [inline snippets/reduceWhich.md]: The example here is incorrect
```js
reduceWhich([{name: 'Tom', age: 12}, {name: 'Jack', age: 18}, {name: 'Lucy', age: 9}], function(a, b){return b.age - a.age;}) // {name: "Jack", age: 18}
```

Comment (liuliangsir) [reply snippets/reduceWhich.md]: Fix

File: snippets/reduceWhich.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+### reduceWhich
+
+By setting some rules, get Minimum or Maximum value in array .
```
Comment (Chalarangelo) [inline snippets/reduceWhich.md]: The description should explain what the snippet method is used for.

File: snippets/reduceWhich.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+BTW, the Minimum / Maximum value could be `Object`, namely, the upper array could be a `Object` array(filled with `Object`).
```
Comment (Chalarangelo) [inline snippets/reduceWhich.md]: The explanation of the snippet should briefly mention native methods and techniques used to achieve this functionality.

File: snippets/reduceWhich.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    return comparator(a, b) >= 0 ? b : a;
+});
+```
+
+```js
+// get minimum value from numeric array
+reduceWhich([1, 3, 2]); // 1
+
+// get maximum value from numeric array
+reduceWhich([1, 3, 2], function (a, b) {return b - a;}); // 3
```
Comment (Chalarangelo) [inline snippets/reduceWhich.md]: All functions in examples should be converted to arrow functions.

File: snippets/reduceWhich.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+BTW, the Minimum / Maximum value could be `Object`, namely, the upper array could be a `Object` array(filled with `Object`).
+
+```js
+const reduceWhich = (xs, comparator = (a, b) => a - b) => xs.reduce(function (a, b) {
+    return comparator(a, b) >= 0 ? b : a;
+});
+```
+
+```js
+// get minimum value from numeric array
```
Comment (Chalarangelo) [inline snippets/reduceWhich.md]: Single line comments above examples should be removed and instead added in parentheses next to the result in the inline comment of each example (if necessary).

File: snippets/reduceWhich.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+reduceWhich([1, 3, 2]); // 1
+
+// get maximum value from numeric array
+reduceWhich([1, 3, 2], function (a, b) {return b - a;}); // 3
+
+// get minimum value from object array by a given key map to value
+reduceWhich([{name: 'Tom', age: 12}, {name: 'Jack', age: 18}, {name: 'Lucy', age: 9}], function(a, b){return a.age - b.age;}) // {name: "Lucy", age: 9}
+
+// get maximum value from object array by a given key map to value
+reduceWhich([{name: 'Tom', age: 12}, {name: 'Jack', age: 18}, {name: 'Lucy', age: 9}], function(a, b){return b.age - a.age;}) // {name: "Jack", age: 18}
```
Comment (Chalarangelo) [inline snippets/reduceWhich.md]: The above examples are sufficient to showcase the functionality of the snippet, this last example should be removed.

File: snippets/reduceWhich.md (hunk 4 above)
Comment (liuliangsir) [reply snippets/reduceWhich.md]: Fixed

File: snippets/reduceWhich.md (hunk 6 above)
Comment (liuliangsir) [reply snippets/reduceWhich.md]: Removed

File: snippets/reduceWhich.md (hunk 5 above)
Comment (liuliangsir) [reply snippets/reduceWhich.md]: Removed

File: snippets/reduceWhich.md (hunk 3 above)
Comment (liuliangsir) [reply snippets/reduceWhich.md]: Fixed

File: snippets/reduceWhich.md (hunk 2 above)
Comment (liuliangsir) [reply snippets/reduceWhich.md]: Fixed

**Pull request merged**